evalml/tests/pipeline_tests/test_pipelines.py

```diff
@@ -2754,3 +2755,80 @@ def test_pipeline_transform_with_final_estimator(
         ),
     ):
         pipeline.transform(X, y)
+
+
+@patch("evalml.pipelines.components.LogisticRegressionClassifier.fit")
+def test_training_only_component_in_pipeline_fit(mock_fit):
+    X = pd.DataFrame(
+        {
+            "a": [i for i in range(9)] + [np.nan],
+            "b": [i % 3 for i in range(10)],
+            "c": [i % 7 for i in range(10)],
+        }
+    )
+    y = pd.Series([0] * 5 + [1] * 5)
```

Comment:
Admittedly this is on my backburner's backburner - but are able to either use an existing X/y pair for these tests or create a module level test fixture for these additional tests?

Reply:
Thank you for holding me accountable, done!